setup.py

```diff
@@ -35,6 +35,7 @@
     ],
 
     install_requires=['jsonpickle', 'wrapt', 'requests'],
+    extras_require={'async': ['tasklocals']},
```

Comment:
Could you provide more information about tasklocals? Its GitHub repo doesn't have a license and it seems like a dead project for almost 4 years so I don't think it is proper to use it as a dependency.

Reply:
Yeah fair point. I mean its not that hard to re-implement I was just thinking there wasn't a point in reinventing the wheel. In this case I think I'll just implement task local storage directly into async_context.py